### ISSUE TITLE
Add Dot layer

### ISSUE THREAD

**Issue by mkaze** [JetBrains/KotlinDL#144]
Currently, the support for the Dot layer is missing and it would be great if we could add support for this layer as well. The desired PR addressing this issue should include:

- [ ] Implementation of layer class named as `Dot` inherited from the [AbstractMerge ](https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/merge/AbstractMerge.kt) class
- [ ] Documentation of layer.
- [ ] Test cases
- [ ] Support for export of layer to JSON (see [ModelSaver.kt][1])
- [ ] Support for import of layer from JSON (see [ModelLoader.kt][2])

Also, if needed, you can take a look at [Keras documentation for this layer](https://keras.io/api/layers/merging_layers/dot/). 

[1]: https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/ModelSaver.kt

[2]: https://github.com/JetBrains/KotlinDL/blob/master/api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/inference/keras/ModelLoader.kt

**Comment by zaleslaw** [comment]
Good issue, we could put it on the roadmap

**Comment by justinyue1643** [comment]
My friend, @RandyHuynh5815, and I would like to take a shot at this task.

**Comment by zaleslaw** [comment]
Hi! @justinyue1643, I assigned it to you

**Comment by zaleslaw** [comment]
Hi, @justinyue1643. Are you going to continue to work on this issue? It will be moved to the 0.4 milestone.

**Comment by therealansh** [comment]
Hey @zaleslaw , I could try this one! Does the `Ops` right now have `batch_dot`? Also i could not find `l2_normalize` for normalizing.

**Comment by zaleslaw** [comment]
Unfortunately, I could not find these operations, probably `BatchMatMulV2 `could be used instead of `batch_dot`?

l2_normalize is missed because it's implemented in Python fully, we need to implement it in a KotlinDL like [here](https://github.com/tensorflow/tensorflow/blob/v1.15.0/tensorflow/python/ops/nn_impl.py#L620)

For example, I've created an implementation of l2_normalize below
`fun  l2Normalize(scope: Scope?, x: Operand<Float>, axis: IntArray?): Operand<Float>? {
    val squareSum: Operand<Float> = ReduceSum.create(
        scope,
        Square.create(scope, x),
        Constant.create(scope, axis),
        ReduceSum.keepDims(Boolean.TRUE)
    )
    val invNorm: Operand<Float> = Rsqrt.create(
        scope,
        org.tensorflow.op.math.Maximum.create(
            scope, squareSum, Constant.create(scope, 1e-12f)
        )
    )
    return Mul.create(scope, x, invNorm)
}`
